pkg/operator/certrotation/target_test.go

```diff
@@ -631,3 +636,217 @@ func TestEnsureTargetSignerCertKeyPair(t *testing.T) {
+
+	firstSigner, err := crypto.GetCAFromBytes(firstSignerSecret.Data["tls.crt"], firstSignerSecret.Data["tls.key"])
+	require.NoError(t, err)
+	secondSigner, err := crypto.GetCAFromBytes(secondSignerSecret.Data["tls.crt"], secondSignerSecret.Data["tls.key"])
+	require.NoError(t, err)
+	thirdSigner, err := crypto.GetCAFromBytes(thirdSignerSecret.Data["tls.crt"], thirdSignerSecret.Data["tls.key"])
+	require.NoError(t, err)
+
+	fakeKubeClient := kubefake.NewSimpleClientset()
+	coreV1Indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+
+	target := CABundleConfigMap{
+		Namespace:     "ns",
+		Name:          "trust-bundle",
+		Client:        fakeKubeClient.CoreV1(),
+		Lister:        corev1listers.NewConfigMapLister(coreV1Indexer),
+		EventRecorder: events.NewInMemoryRecorder("test"),
+	}
+
+	expectedCertificates := []*x509.Certificate{firstSigner.Config.Certs[0]}
+	res, err := target.EnsureConfigMapCABundle(context.TODO(), firstSigner)
+	require.NoError(t, err)
+	require.Equal(t, expectedCertificates, res)
+	caBundleConfigMap, err := fakeKubeClient.CoreV1().ConfigMaps("ns").Get(context.Background(), "trust-bundle", metav1.GetOptions{})
+	require.NoError(t, err)
+	coreV1Indexer.Add(caBundleConfigMap)
+
+	res, err = target.EnsureConfigMapCABundle(context.TODO(), secondSigner)
+	require.NoError(t, err)
+	// by not adding the raBundleConfigMap
+	// to the lister, we simulate the race condition
+	// that causes overwriting of the CM
+	//
+	// nevertheless, let's add the secondSigner to our expectedCertificates
+	expectedCertificates = append(expectedCertificates, secondSigner.Config.Certs[0])
+	sort.SliceStable(expectedCertificates, func(i, j int) bool {
+		return bytes.Compare(expectedCertificates[i].Raw, expectedCertificates[j].Raw) < 0
+	})
+	require.Equal(t, expectedCertificates, res)
+
+	res, err = target.EnsureConfigMapCABundle(context.TODO(), thirdSigner)
+	require.NoError(t, err)
+	expectedCertificates = append(expectedCertificates, thirdSigner.Config.Certs[0])
+	sort.SliceStable(expectedCertificates, func(i, j int) bool {
+		return bytes.Compare(expectedCertificates[i].Raw, expectedCertificates[j].Raw) < 0
+	})
+	require.Equal(t, expectedCertificates, res)
+}
+
+func TestRotatedSigningCASecretSameCommonName(t *testing.T) {
+	fakeKubeClient := kubefake.NewSimpleClientset()
+	coreV1Indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+
+	target := RotatedSigningCASecret{
+		Namespace:     "ns",
+		Name:          "signer",
+		Validity:      24 * time.Hour,
+		Refresh:       12 * time.Hour,
+		Client:        fakeKubeClient.CoreV1(),
+		Lister:        corev1listers.NewSecretLister(coreV1Indexer),
+		EventRecorder: events.NewInMemoryRecorder("test"),
+	}
+
+	firstSigner, signerUpdated, err := target.EnsureSigningCertKeyPair(context.TODO())
+	require.NoError(t, err)
+	require.True(t, signerUpdated)
```

Comment:
We don't cache signer here. If a proper caching is added this will no longer generate different commonName signer:
```
	signerSecret := &corev1.Secret{
		ObjectMeta: metav1.ObjectMeta{
			Name:      "signer",
			Namespace: "ns",
		},
	}
	ensureMetadataUpdate(signerSecret, signer.Owner, signer.AdditionalAnnotations)
	err = setSigningCertKeyPairSecret(signerSecret, signer.Validity)
	require.NoError(t, err)
	coreV1Indexer.Add(signerSecret)
```